Updated App install banners samples for #128

PaulKinlan opened the pull request.

PaulKinlan committed:
Padding

PaulKinlan committed:
Merge branch 'gh-pages' of github.com:GoogleChrome/samples into gh-pages

PaulKinlan committed:
Fixes #128 - Updates app install banner sample and adds new ones

* Now supports cancelable event sample
* Added User response demo - i.e, did the user install or dismiss the banner
* Moves basic sample around

PaulKinlan commented:
cc @addyosmani @jeffposnick

jeffposnick commented:
Can you add a reference to https://github.com/GoogleChrome/samples/blob/gh-pages/styles/main.css to all three `index.html` files?

jeffposnick commented on app-install-banner/basic-banner/index.html:
```diff
@@ -17,11 +17,14 @@
   <head>
+    <title>Web App Install Banner: Basic Prompt Sample</title>
     <meta name="viewport" content="width=device-width, initial-scale=1">
     <link rel="manifest" href="./manifest.json">
```
Simplify `href="./manifest.json"` to `href="manifest.json"`

jeffposnick commented on app-install-banner/cancelable-banner/index.html:
```diff
@@ -0,0 +1,69 @@
+  <head>
+    <title>Web App Install Banner: Cancel Prompt Sample</title>
+    <meta name="viewport" content="width=device-width, initial-scale=1">
+    <link rel="manifest" href="./manifest.json">
```
Simplify `href="./manifest.json"` to `href="manifest.json"`

jeffposnick commented on app-install-banner/useraction-banner/index.html:
```diff
@@ -0,0 +1,80 @@
+  <head>
+    <title>Web App Install Banner: Detect Install or Cancel Prompt Sample</title>
+    <meta name="viewport" content="width=device-width, initial-scale=1">
+    <link rel="manifest" href="./manifest.json">
```
Simplify `href="./manifest.json"` to `href="manifest.json"`

jeffposnick commented on app-install-banner/cancelable-banner/index.html:
```diff
@@ -0,0 +1,69 @@
+<!DOCTYPE html>
+<!--
+Copyright 2014 Google Inc. All Rights Reserved.
```
2015

jeffposnick commented on app-install-banner/useraction-banner/index.html:
```diff
@@ -0,0 +1,80 @@
+<!DOCTYPE html>
+<!--
+Copyright 2014 Google Inc. All Rights Reserved.
```
2015

jeffposnick commented on autocapitalize/index.html:
```diff
@@ -42,6 +42,11 @@
 
     <link rel="stylesheet" href="../styles/main.css">
     <style>
+
```
Did you mean to include changes to this sample along with this PR?

PaulKinlan replied:
no....

jeffposnick commented on app-install-banner/cancelable-banner/index.html:
```diff
@@ -0,0 +1,69 @@
+      <li>The app uses a service worker</li>
+      <li>The site is using SSL</li>
+      <li>The app has a manifest declared</li>
+      <li>The manifest has a `short_name`, 144 pixel icon and a type of 'image/png'</li>
```
`short_name` → &lt;code>short_name&lt;/code>

jeffposnick commented on app-install-banner/useraction-banner/index.html:
```diff
@@ -0,0 +1,80 @@
+      <li>The app uses a service worker</li>
+      <li>The site is using SSL</li>
+      <li>The app has a manifest declared</li>
+      <li>The manifest has a `short_name`, 144 pixel icon and a type of 'image/png'</li>
```
`short_name` → &lt;code>short_name&lt;/code>

jeffposnick commented on app-install-banner/cancelable-banner/index.html:
```diff
@@ -0,0 +1,69 @@
+    <ul>
+      <li>The app uses a service worker</li>
+      <li>The site is using SSL</li>
+      <li>The app has a manifest declared</li>
```
It would be nice to link this to http://updates.html5rocks.com/2014/11/Support-for-installable-web-apps-with-webapp-manifest-in-chrome-38-for-Android

jeffposnick commented on app-install-banner/useraction-banner/index.html:
```diff
@@ -0,0 +1,80 @@
+    <ul>
+      <li>The app uses a service worker</li>
+      <li>The site is using SSL</li>
+      <li>The app has a manifest declared</li>
```
It would be nice to link this to http://updates.html5rocks.com/2014/11/Support-for-installable-web-apps-with-webapp-manifest-in-chrome-38-for-Android

jeffposnick commented on app-install-banner/cancelable-banner/index.html:
```diff
@@ -0,0 +1,69 @@
+    </p>
+
+    <ul>
+      <li>The app uses a service worker</li>
```
It would be nice to link this to http://www.html5rocks.com/en/tutorials/service-worker/introduction/

jeffposnick commented on app-install-banner/useraction-banner/index.html:
```diff
@@ -0,0 +1,80 @@
+    </p>
+
+    <ul>
+      <li>The app uses a service worker</li>
```
It would be nice to link this to http://www.html5rocks.com/en/tutorials/service-worker/introduction/

jeffposnick commented on app-install-banner/cancelable-banner/index.html:
```diff
@@ -0,0 +1,69 @@
+
+    <ul>
+      <li>The app uses a service worker</li>
+      <li>The site is using SSL</li>
```
SSL → HTTPS

jeffposnick commented on app-install-banner/useraction-banner/index.html:
```diff
@@ -0,0 +1,80 @@
+
+    <ul>
+      <li>The app uses a service worker</li>
+      <li>The site is using SSL</li>
```
SSL → HTTPS

jeffposnick commented on app-install-banner/useraction-banner/index.html:
```diff
@@ -0,0 +1,80 @@
+    
+    <pre id=output></pre>
+
+    <p>To see if this example works, come back in another day and you will see the banner.</p>
```
"in another day" implies that I can come back tomorrow and I'd see the banner. Given the delay between visits that's necessary to trigger the banner, I think we need to strongly push folks to chrome://flags if they want to try this out.

jeffposnick commented on app-install-banner/useraction-banner/readme.md:
```diff
@@ -0,0 +1,13 @@
+App Install Banner User Response Sample
+===
+
+See https://googlechrome.github.io/samples/app-install-banner/index.html for a live demo.
```
https://googlechrome.github.io/samples/app-install-banner/useraction-banner/index.html

jeffposnick commented:
Thanks for making these updates, Paul! There are a few inline comments that apply to all three `index.html`s if you keep the current structure. (I probably should have caught some of them from the initial review.)

I'm wondering whether you want to consider structuring this slightly differently—instead of moving the existing https://github.com/GoogleChrome/samples/blob/gh-pages/app-install-banner/index.html file to a `basic-banner` directory (and potentially breaking any sites that link to that old URL), what about just adding in the functionality from the `useraction-banner` sample into the existing file? Then you can do away with the `useraction-banner`directory.

Keeping the new `canceable-banner` in a separate directory is fine by me.

jeffposnick commented on app-install-banner/useraction-banner/index.html:
```diff
@@ -0,0 +1,80 @@
+      window.addEventListener('beforeinstallprompt', function(e) {
+        outputElement.textContent = 'beforeinstallprompt Event fired';
+        
+        e.userChoice.then(function(choiceResult) {
```
Can you put a comment here about how `userChoice` is a Promise and link to something like http://www.html5rocks.com/en/tutorials/es6/promises/ for those who aren't familiar?

PaulKinlan replied:
Not a bad idea, but I didn;t think we were annotating code this way.  will get on it.

PaulKinlan commented:
Hmm, I could do.  I did what I did so that in theory links would still point to a root directory and I can disambiguate there (which I think I should sort out).

I'll deal with all the other comments first.

PaulKinlan committed:
Fixing more issues with app install banner samples

PaulKinlan committed:
Adding stylesheet to app install banner samples

PaulKinlan commented:
@jeffposnick a bunch of updates:
- added a new index file for all existing links
- fixed comments
- can't undo the autocaptilaize (which needs to be pushed up anyway).

jeffposnick commented on app-install-banner/cancelable-banner/index.html:
```diff
@@ -0,0 +1,70 @@
+    </ul>
+    
+    <p>In this example you will see below a message indicating that the beforeinstallprompt occured (only after meeting the above criteria) and was cancelled.</p>
+    
```
Can you add a sentence here explaining why this is desirable? E.g. "If your app already displays its own prompt to add to homescreen..." or something along those lines.

jeffposnick commented on app-install-banner/index.html:
```diff
@@ -17,39 +17,31 @@
-    It will only prompt when a number of criteria have been met:
+    <h1>Web App Install Banner Samples</h1>
+	
+    <p>The w<a href="http://updates.html5rocks.com/2014/11/Support-for-installable-web-apps-with-webapp-manifest-in-chrome-38-for-Android">web app install banner</a> user prompt that Chrome will trigger to indicate that the user 
```
Stray 'w' character before the link.

jeffposnick commented:
Okay, looks good in general, and I don't have objections to creating an index that points to the other samples. I had two additional comments after going over the revised code.

PaulKinlan committed:
Fixing some final changes for the app install banner.  Clearing up the wording.

jeffposnick commented:
:+1:
